vars/signArtifacts.groovy

```diff
@@ -24,11 +26,21 @@ void call(Map args = [:]) {
             export UNSIGNED_BUCKET=${SIGNER_CLIENT_UNSIGNED_BUCKET}
             export SIGNED_BUCKET=${SIGNER_CLIENT_SIGNED_BUCKET}
 
-            $WORKSPACE/sign.sh ${args.artifactPath} --sigtype=${args.signatureType} --component=${args.component} --type=${args.type}
+            $WORKSPACE/sign.sh ${arguments}
         """
     }
 }
 
+String generateArguments(args) {
+    String arguments = args.artifactPath
+    for (entry in args) {
+        if (!entry.key.equals("artifactPath")) {
+            arguments += " --${entry.key}=${entry.value}"
```

Comment:
entry.key would not work with `--sigType`

Reply:
We didn't have a unified name for the signature type here. According to (https://github.com/opensearch-project/opensearch-build/blob/f3d8bcda4bc015786a7c167e1563fb22632ff763/src/run_sign.py#L26) we should then change the related Jenkins files arguments with arg like `sigtype: .sig` instead of `signatureType` as before.

Reply:
I think changing `opensearch-build/src/run_sign.py` would be better since `signArtifacts` is being called from a lot of places, and would require to you to test out a lot of jobs.

Reply:
That makes sense to me. Changed.

Reply:
We can also remove the function and do the following - 

```groovy

// Since the first argument should be artifactPath
String arguments = args.artifactPath

args.each(entry -> entry.key.equals('artifactPath') ? "" : arguments += " --${entry.key}=${entry.value}")

```

Reply:
adding this has only one issue, it's hard to recognize what arguments does this function take. I think it would be a good idea to add a javadoc like comment on top describing the params. WDYT?

Reply:
Looks like out of scope changes. @zelinh Maybe refactoring come in later PR and this PR should concentrate on maven job only?

Reply:
The maven job would require these changes to run with opensearch-build setup.

Reply:
We could add a description comment on top of this file for the instruction of this library which will tell people how the acceptable arguments.

Reply:
This change feels over complicated. Why add a loop to get all the arguments when you can get it directly form `args.variable`or maybe I am misunderstanding something?

Reply:
I was trying to iterating through the `args` without knowing what arguments would be passed in. So use the loop to iterate through the keys of the `args` map.